### ISSUE TITLE
Tags are ignored for constants defined via 'enum'

### ISSUE THREAD

**Issue by m1n1vv** [pawn-lang/compiler#444]
### Issue description:
Tags seem to be ignored for constants defined through an enumeration.

### Minimal complete verifiable example (MCVE):
The following code
```Pawn
#include <a_samp>

enum
{
	Float:FLOAT_VALUE = 0.64
};

main()
{
	new Float:f = FLOAT_VALUE;
	printf("f = %.2f", f);
}
```
outputs `f = 1059313408.00` instead of `f = 0.64`, which means for some reason "FLOAT_VALUE" has tag "_" instead of "Float", so an implicit conversion takes place (something like `float(_:0.64)`).

### Workspace Information:

* Compiler version: 3.10.9, 3.2.3664
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
Another example:
```Pawn
#include <float>

enum { MAX_AIR_OBJECTS = 4 };
enum { Float:SPAWN_DISTANCE_AREA = 1500.0 };

main()
{
	new x, Float:f;
	#pragma unused f, x
	#assert tagof(MAX_AIR_OBJECTS) == tagof(x)
	#assert tagof(SPAWN_DISTANCE_AREA) == tagof(f)
}
```
For this code the compiler would issue the following error:
```
test.pwn(12) : fatal error 110: assertion failed: tagof(SPAWN_DISTANCE_AREA) == tagof(f)
```
Which means that the tag of the `SPAWN_DISTANCE_AREA` constant differs from `Float`.
Interestingly, this issue doesn't occur on the stock version of the compiler (3.2.3664).

**Comment by m1n1vv** [comment]
```pawn
static enum 
{ 
    MAX_AIR_OBJECTS = 4 
}; 

static enum Float: 
{ 
    SPAWN_DISTANCE_AREA = 1500.0 
};
```

**Comment by Daniel-Cortez** [comment]
In addition to the 1'st post, defining `FLOAT_VALUE` in the following way
```Pawn
enum Float:
{
	FLOAT_VALUE = 0.64
};
```
fixes the issue, but it doesn't cancel the problem that the compiler silently ignores all tags inside `{ ... }`.
I think it would be reasonable to take in account tags inside the braces, or issue errors for them if the user has already specified a tag before the opening brace.
Example:
```Pawn
enum
{
	INT_VALUE = 0,
	Float:FLOAT_VALUE = 1.0 // FLOAT_VALUE has tag 'Float', not '_'
};

enum eCUSTOM_TAG:
{
	VALUE1 = 0,
	Float:VALUE2 = 1 // error (the user has already specified tag 'eCUSTOM_TAG'
	                 // which is applied to all constants within the enumeration)
};
```

**Comment by Y-Less** [comment]
That would start breaking on arrays.  This:

```pawn
enum X
{
    A,
    Float:B
}

new arr[X];

printf("%f", arr[B]);
```

Is just sugar for:

```pawn
#define A (X:0)
#define B (X:1)

new arr[X:2];

printf("%f", Float:arr[B]);
```

If we apply the tag to the constant directly we get an index tag mismatch:

```pawn
#define A (X:0)
#define B (Float:1)

new arr[X:2];

printf("%f", arr[B]);
```

They're the same constants used in different places, here the numbers just happen to be used as an index.

**Comment by Daniel-Cortez** [comment]
I meant for this new rule to apply only to unnamed (anonymous) enumerations, not the named ones (sorry for not being clear in the above message). While this rule might break named enumerations, it doesn't have to apply to them, the problem is that tags are silently ignored in the declarations of **unnamed** enums.

**Comment by Y-Less** [comment]
I just realised that you can't declare an array using an anonymous enum anyway, so that restriction would be guaranteed to not break arrays at least.

The only question then becomes:

```pawn
enum (<<= 1)
{
    Float:A = 4.3,
    B
}
```

What is `B`?

**Comment by Daniel-Cortez** [comment]
> What is `B`?

Since the `Float` tag in your example is ignored (so `A` is `_:4.3`), `B` would be `_:4.3 + 1`.

**Comment by Y-Less** [comment]
Yes, but the point is we're talking about not ignoring the tags.

**Comment by Y-Less** [comment]
There are two ways to do it:

1) Rounding.  So `int(4.3) << 1`.  Which rounding method to use is another point - floor, to-zero, nearest, banker, etc.

2) Bit.  So `_:4.3 << 1`.

The former would give `8` in this example, though not always that obvious based on the rounding method used.  Should `3.9` round to `3` or `4`?

The latter would give `0b10000001000100110011001100110100` (-2129448140), which is the binary representation of the float left shifted by 1.

**Comment by Daniel-Cortez** [comment]
I don't think I get your point. Returning to that example with `A` and `B`, with the restriction I proposed the compiler would issue an error (because the `Float` tag has no effect inside the enum braces), so why would it matter what is `B` at all?

**Comment by Y-Less** [comment]
Apparently I totally mis-read your earlier post.

**Comment by Daniel-Cortez** [comment]
@pawn-lang Please close this (fixed in #454).
